index.js

```diff
@@ -56,26 +54,46 @@ export function checkDataIntegrityProofFormat({
             'Expected proof to be either an object or an array.'
           );
         });
-        it('`type` field MUST exist and be a string.', function() {
+        it('if "id" field exists, it MUST be a valid URL.', function() {
           this.test.cell = {columnId: vendorName, rowId: this.test.title};
           for(const proof of proofs) {
-            proof.should.have.property('type');
-            proof.type.should.be.a('string');
+            if(proof.id) {
+              let result;
+              let err;
+              try {
+                result = new URL(proof.id);
+              } catch(e) {
+                err = e;
+              }
+              should.not.exist(err, 'Expected URL check of the "proof.id" ' +
+                'to not error.');
+              should.exist(result, 'Expected "proof.id" to be a URL.');
+            }
           }
         });
-        it('`type` field MUST be `DataIntegrityProof`.', function() {
+        it('"type" field MUST exist and be a string.', function() {
           this.test.cell = {columnId: vendorName, rowId: this.test.title};
           for(const proof of proofs) {
             proof.should.have.property('type');
             proof.type.should.be.a(
-              'string',
-              'Expected `proof.type` to be a string.'
-            );
-            proof.type.should.equal('DataIntegrityProof');
+              'string', 'Expected `proof.type` to be a string.');
+          }
+        });
+        it('"type" field MAY be either "DataIntegrityProof" or ' +
```

Comment:
Is this correct? In the Verifiable Credential Data Integrity 1.0 spec  under section `2.1 Proofs` it states that  examples of proof types include DataIntegrityProof and Ed25519Signature2020 whereas under section `3.1 DataIntegrityProof` it states that the type property MUST contain the string `DataIntegrityProof`. Not sure if one of them is incorrect or if both are correct and we require separate assertion function `checkDataIntegrityProof` for the crypto suites `3.1 DataIntegrityProof`.

Reply:
Not clear from the PR title which version you are trying to test.

Afaik, in v2 only `DataIntegrityProof` is legal RDF call for proof predicate.

I guess it's always possible you have multiple inheritance though, so you might want to add a test for [ DataIntegrityProof, Ed25519Signature2020 ]

Reply:
@OR13 I was referring to version 1.0 https://www.w3.org/TR/vc-data-integrity/, didn't know 2.0 existed. I'll take a look at that, thanks.

Reply:
@msporny you might need to clear up this confusion. I believe we are moving away from `Ed25519Signature2020` to just `DataIntegrityProof`, but the spec still lists both values as valid in places.

Reply:
> @msporny you might need to clear up this confusion. I believe we are moving away from `Ed25519Signature2020` to just `DataIntegrityProof`, but the spec still lists both values as valid in places.

`Ed25519Signature2020` is widely deployed for the TruAge program, and has multiple demonstrated implementations as of the last JFF Plugfest #2, so we are documenting that because it exists in the wild. We are not putting that type in the core v2 context, if developers want to use it, you have to include another context file (which is what all of the TruAge implementations do today).

`DataIntegrityProof` is what is in the core v2 context and is what the majority of the Data Integrity cryptosuites use (in fact, every one of them, except for `Ed25519Signature2020` use it at present).

So, both are valid types, but the tests for the specific type should be moved to each cryptosuite and shouldn't be placed in this test suite assertion library.